add A record slave.<domain> to contain list of slave IPs

lloesche opened the pull request:
This allows me to resolve slave.mesos and get a list of all slave IPs.
We'd like to use this for our dreamathon monitoring project which is supposed to run on DCOS and monitor the state of all mesos slaves.

In a second step I'd also like to add appropriate SRV records, but couldn't find an easy way to get the slave port out of the current data structure.

jdef commented:
the port shows up in two places in the slave state.json: flags.port and in pid. it's probably pretty simple to add a "pid" field to the slave struct, and then parse the port from that. sample data for pid:
```
slave(1)@10.2.0.5:5051
```

you can use mesos-go/upid to parse if you want (mesos-go is already a dep):
https://github.com/mesos/mesos-go/blob/master/upid/upid.go

lloesche commented:
@jdef `go test ./...` now passes but the gometalinter throws lots of warnings that don't seem related to my patch?!

tsenart commented:
@lloesche: You can ignore those gometalinter failures for now.

Regarding your patch, it's great that you fixed the test but as far as I can see, the added functionality needs its own test.

lloesche commented:
@tsenart added. I'm learning so much about Go today :-)

tsenart commented:
It's fun isn't it?! :-)

tsenart commented:
I'll take another look at this once I get back home.

tsenart commented on records/generator_test.go:
```diff
@@ -287,6 +287,14 @@ func TestInsertState(t *testing.T) {
+		t.Error("should find running slaves - A record")
+	}
+	if got, want := rrs, []string{"1.2.3.10", "1.2.3.11", "1.2.3.12"}; !reflect.DeepEqual(got, want) {
+		t.Errorf("should return the slave ips 1.2.3.10, 1.2.3.11, 1.2.3.12 for the slaves record, but got %v", rrs)
```
Can we not hardcode the wanted value in this string?
```go
t.Errorf("got: %v, want: %v", got, want)
```

tsenart commented:
Apart from the small comment, LGTM. Please squash all commits into one once you're done.

lloesche commented:
Removed the hard coded IPs from the test (a pattern which I by the way copied from other tests :-)) and squashed everything into one commit. Also I updated the name to be singular slave.mesos instead of slaves.mesos to be consistent with master.mesos.

tsenart commented:
LGTM. @lloesche please open a new issue to track:
> In a second step I'd also like to add appropriate SRV records, but couldn't find an easy way to get the slave port out of the current data structure.

jdef commented on records/generator.go:
```diff
@@ -271,12 +271,15 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
+
 	for _, slave := range sj.Slaves {
-		rg.SlaveIPs[slave.ID] = sanitizedSlaveAddress(slave.Hostname, spec)
+		ssa := sanitizedSlaveAddress(slave.Hostname, spec)
```
actually there's a chance that `sanitizedSlaveAddress` doesn't return a string containing an ipv4 address, which will result in a bad A record. should probably test for that.

lloesche replied:
@jdef does that affect rg.SlaveIPs[slave.ID] as well? or is it okay for rg.SlaveIPs to contain invalid IPs? I.e. should I only check for the part where I add an A record for the slave.domain name or also for the rg.SlaveIPs map?

lloesche replied:
@jdef fixed in https://github.com/mesosphere/mesos-dns/pull/199
